### ISSUE TITLE
USWDS - Bug: Disabled color settings do not check for contrast

### ISSUE THREAD

**Issue by amyleadem** [uswds/uswds#5427]
## Summary
When customizing the disabled state color settings, the system does not send a warning when the selected colors do not meet required contrast ratios. 

For example, the following configuration results in text that does not meet color contrast requirements with its background :

```
@use "uswds-core" with (
  $theme-show-notifications: false,
  $theme-color-disabled-lighter: "gray-20",
  $theme-color-disabled-light: "gray-20",
  $theme-color-disabled: "gray-20",
  $theme-color-disabled-dark: "gray-20",
  $theme-color-disabled-darker: "gray-20",
);
```

Here is a screenshot of the resulting display (note that the text is not visible in the button and select components and that the outline button does not meet contrast requirements):
![image](https://github.com/uswds/uswds/assets/93996430/586d8251-e70c-4068-9e87-d36599cc3f9e)

In contrast, here is what it looks like in the default configuration:
![image](https://github.com/uswds/uswds/assets/93996430/d4695918-b1e4-4b0a-b769-91045f4d0675)

## Expected behavior
The system should send a warning when settings values do not result in the required color contrast ratio.

**Comment by mejiaj** [comment]
Moving to next release (3.8.0) as a potential enhancement.

**Comment by maggiewachs** [comment]
@amyleadem @mahoneycm Hello! I'm a web accessibility engineer with CivicActions; we have a few projects that use the USWDS. 

I was wondering what's the reason for this fix? Disabled elements [don't need to meet contrast minimum](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html), and that's by design because they're meant to look inactive/unusable. Making them higher contrast could actually be confusing to users, especially those with low vision.

**Comment by jaclinec** [comment]
@maggiewachs Thanks for reaching out!

Over the last several months, we have been researching and re-evaluating the role of disabled elements in USWDS. Our research pointed us to pursue the following actions:

1) Draft recommendations against using disabled elements at all (this is a work in progress), and 
2) Update the styles of USWDS disabled elements to meet color contrast requirements. 

We updated our disabled styles because our research pointed to disabled elements having meaning. Including disabled elements on a page indicates to users that an action must be taken before the element can be enabled. By providing styles that failed to meet color contrast requirements, we were excluding certain audiences from being able to access and understand that meaning. We understand that this deviates from the low-contrast presentation that is the common standard, but it felt important to include all audiences here.

We’re drafting guidance that recommends offering additional feedback (not just a visual cue) to let users know why something is disabled, such as a helper text or a tooltip. We're also drafting guidance for best practices for coding disabled elements for screen readers.

That said, we are also working on providing recommendations for moving away from disabled elements entirely (with a few exceptions). Instead we recommend projects structure their content in a way that only offers actionable elements to users. We aim to have this guidance out in the next few sprints!

**Comment by adriancooke** [comment]
Hi @jaclinec, I’m a UX designer, also at CivicActions. 👋 Thanks very much for detailing some of the background work and thinking in response to @maggiewachs’s question. 

We were looking at adopting the color contrast update for disabled form elements on a project when we realized that doing so without the additional feedback you mentioned might actually confuse users by making disabled components seem more prominent and therefore decreasing the visual differences between enabled and disabled elements. 

Understanding that your work on this is part of forthcoming recommendations about how and whether to use disabled elements, I wonder if you’d consider providing interim advice (and maybe even caution) on adopting the increased contrast update by itself?

**Comment by maggiewachs** [comment]
@jaclinec thanks for the reply! I've read through the issues that outline your research, and it's really thorough. It's really nice to see accessibility given such priority.

If you have time, do you have any background information on why a gray color was chosen and not, for example, a shift in opacity that is at or near the contrast value? Thinking of how colorblind users may interpret the disabled state: 
![image](https://github.com/uswds/uswds/assets/228885/f91200b2-0cfa-43d4-be8f-177604c5cd93)

**Comment by jaclinec** [comment]
@adriancooke Yes there is potential that it could cause confusion without the additional feedback, so we’re hoping to get that guidance out soon. It’s a bit of a seesaw - fixing an issue for one user group could negatively impact another, and many times the right answer is “it depends.” If you must use disabled elements, consider using an additional mechanism to communicate that it is disabled if at all possible. 

@maggiewachs With the opacity, we could not guarantee proper color contrast. Adding to the complexity is that [WCAG doesn’t provide guidance on how to measure contrast for semi-transparent colors](https://webaim.org/articles/contrast/#:~:text=Gradients%2C%20background%20images%2C%20and%20transparencies). But I think it’s a good question, and one that I’d like to dig into more and see if it's something to add to our research findings and guidance on disabled styles. We provided these styles as a failsafe to ensure uniformity across browsers (and this was the initial issue we faced that spurred up all this research).

Thinking of how someone with monochrome color blindness may experience a disabled button I think is further reason to consider that disabled styles alone are not an ideal choice and it is better to allow the user to interact with elements and offer guidance on why the button isn't clicking. And as always, we recommend testing with your users to see what the best solution is.

I hope some of this helps!
